### PR TITLE
Ladybird/Qt: Resize the viewport when resize is requested

### PR DESCRIPTION
**Pull request by tcl3**
Previously, when a window resize was programmatically requested through WebDriver, we would update the size of the entire window. We now ensure that the portion of the window containing the WebView is set to the size requested.

I haven't updated the AppKit chrome to replicate this behavior in this PR.

Fixes: http://wpt.live/infrastructure/reftest/size.html

### CODE REVIEW

File: Ladybird/Qt/Tab.cpp [hunk 1]
```diff
@@ -390,8 +390,9 @@ Tab::Tab(BrowserWindow* window, WebContentOptions const& web_content_options, St
     };
 
     view().on_resize_window = [this](auto const& size) {
-        m_window->resize(size.width(), size.height());
-        return Gfx::IntSize { m_window->width(), m_window->height() };
+        auto new_window_size = m_window->size() - m_view->size() + QSize { size.width(), size.height() };
+        m_window->resize(new_window_size);
+        return Gfx::IntSize { m_view->width(), m_view->height() };
```
Comment (tcl3) [inline Ladybird/Qt/Tab.cpp:395]: I've changed this function to resize the viewport to the given size rather than the window, but it seems that the caller of this function expects the window to be set to the given size:

https://github.com/LadybirdBrowser/ladybird/blob/3c897e7cf3594f02f559599e1bf28747c9edba13/Userland/Services/WebContent/WebDriverConnection.cpp#L670-L672

I'm not sure what to do here seeing as there are WPT tests that rely on the new behavior. Running the tests in headless mode would probably be the best idea.

File: Ladybird/Qt/BrowserWindow.cpp [hunk 2]
```diff
@@ -1151,6 +1151,7 @@ void BrowserWindow::resizeEvent(QResizeEvent* event)
     QWidget::resizeEvent(event);
 
     for_each_tab([&](auto& tab) {
+        tab.view().set_viewport_size({ static_cast<int>(tab.view().width() * m_device_pixel_ratio), static_cast<int>(tab.view().height() * m_device_pixel_ratio) });
```
Comment (kalenikaliaksandr) [inline Ladybird/Qt/BrowserWindow.cpp:1154]: commit description mentioned programmatic resize request (on_resize_window), but this event is triggered when window is resized by a user interactive, no? probably would be better to have another commit that describes why this change is needed.

File: Ladybird/Qt/Tab.cpp [hunk 3]
```diff
@@ -390,8 +390,9 @@ Tab::Tab(BrowserWindow* window, WebContentOptions const& web_content_options, St
     };
 
     view().on_resize_window = [this](auto const& size) {
-        m_window->resize(size.width(), size.height());
-        return Gfx::IntSize { m_window->width(), m_window->height() };
+        auto new_window_size = m_window->size() - m_view->size() + QSize { size.width(), size.height() };
```
Comment (kalenikaliaksandr) [inline Ladybird/Qt/Tab.cpp:393]: as I understand `m_window->size() - m_view->size()` needs to be added to the a new size to account for window border painted by Qt, right? would be good to have a comment that explains what is going on here.

**Comment by ADKaster** [comment]
@tcl3 seems we pass that test currently, without this PR. so what's the shtatus of this PR? https://wpt.fyi/results/infrastructure/reftest/size.html?product=ladybird

**Comment by tcl3** [comment]
I think it's probably best to close this. The issue I was trying to solve with this PR ended up being solved by forcing each test to run in a new process.

**Comment by trflynn89** [comment]
The WebDriver spec also explicitly states to resize the chrome window itself:
```
16. Let window be the operating system window containing session's current top-level browsing context

17. If the implementation is able to set the dimensions of window:
    1. If width is not null, set the width, in CSS pixels, of window, including any browser chrome and externally drawn
       window decorations, to a value that is as close as possible to width.
    2. If height is not null, set the height, in CSS pixels, of window, including any browser chrome and externally drawn
       window decorations, to a value that is as close as possible to height. 
```
